### ISSUE TITLE
Unable to mock context in tests

### ISSUE THREAD

**Issue by georgewilde** [Urigo/graphql-modules#443]
When testing a module it is not possible to mock out the context or the providers of imported modules. This causes an issue when the module under test imports other modules that add to the context by making API calls or other side effects.

Here is an example of this:

_example.module.ts_
```typescript
export const ExampleModule = new GraphQLModule({
  imports: [AnotherModule],
  resolvers: exampleResolver,
  typeDefs: exampleTypeDefs,
  context: async (session, currentContext): Promise<ExampleModuleContext> => {
    const { token } = currentContext;
    return Promise.resolve<ExampleModuleContext>({
      token,
    });
  },
});
```

_example.module.spec.ts_
```typescript
describe('ExampleModule', () => {
  it('successfully returns someField', async () => {
    const { schema, injector } = AlertsModule;

    injector.provide({
      provide: ExampleProvider,
      overwrite: true,
      useValue: {
        getSomeField: () => Promise.resolve('example data'),
      },
    });
    
    const result = await execute({
      schema,
      document: gql`
        query {
          someField
        }
      `,
    });
    
    expect(result.data.lastVisitDate).toBe('example data');
  });
});
```

_another.module.ts_
```typescript
export const AnotherModule = new GraphQLModule({
  context: async (session, currentContext, moduleSessionInfo): Promise<ClientRequestContext> => {
    return Promise.resolve({ token: session.req.headers.authorization as string });
  },
});
```

In this example, ExampleModule gets "token" from the currentContext. This is provided by AnotherModule. AnotherModule could get this token from anywhere, in this case it's getting it from the session request headers (which fails when running the test) but it could use AnotherModuleProvider to make a service call etc, all of which is undesirable when wanting to test ExampleModule in isolation.

A way to solve this would be to mock ExampleModule's context and the imported modules. The token in this example could then be substituted with a hard coded string and the imported modules are then not needed for any other reason.

**Comment by ardatan** [comment]
@georgewilde 
Could you try the following code for mocking with this version `0.8.0-alpha.82bbc778`?
```ts
ExampleModule.mock({
 contextBuilder: session => {
        return { token: 'SOMETHING' }
 }
})
```

**Comment by ardatan** [comment]
Fixed in 0.7.6!
